File: data/readable_format/cpp/file1548.cpp
#include <vector>
#include <iostream>
#include <cmath>
#include <cassert>

// Define the COMPRO function directly, without using a class
void COMPRO(std::vector<double>& PROFILE, const std::vector<double>& EMINT, int NFOBR, double W, int JFIRST, int JLAST) {
    for (int K = 0; K < NFOBR; ++K) {
        if (JFIRST == JLAST) {
            PROFILE[K] = EMINT[K];
        } else {
            PROFILE[K] = PROFILE[K] + W * EMINT[K];
        }
    }
}

int main() {
    std::vector<double> profile = {1.0, 2.0, 3.0, 4.0, 5.0};
    std::vector<double> emint = {0.5, 1.5, 2.5, 3.5, 4.5};
    double w = 0.5;
    int jfirst = 1, jlast = 5;
    int NFOBR = 5;

    // Call the function
    COMPRO(profile, emint, NFOBR, w, jfirst, jlast);

    // Expected results based on the original Fortran logic
    std::vector<double> expected = {1.25, 2.75, 4.25, 5.75, 7.25};

    // Assert results
    for (int i = 0; i < NFOBR; ++i) {
        assert(std::fabs(profile[i] - expected[i]) < 1e-9); // Floating-point comparison
    }

    std::cout << "C++ test passed." << std::endl;

    return 0;
}

File: data/readable_format/fortran/file1548.f90
program test_compro
    implicit none
    real*8 :: profile(5), emint(5)
    real*8 :: w
    integer :: jfirst, jlast, i

    ! Initialize test data
    profile = (/1.0, 2.0, 3.0, 4.0, 5.0/)
    emint = (/0.5, 1.5, 2.5, 3.5, 4.5/)
    w = 0.5
    jfirst = 1
    jlast = 5

    ! Call subroutine
    call COMPRO(profile, emint, 5, w, jfirst, jlast)

    ! Output results for manual verification
    do i = 1, 5
        print *, "profile(", i, ") = ", profile(i)
    end do

contains

    subroutine COMPRO(PROFILE, EMINT, NFOBR, W, JFIRST, JLAST)
        implicit none
        real*8, intent(inout) :: PROFILE(:)
        real*8, intent(in) :: EMINT(:)
        real*8, intent(in) :: W
        integer, intent(in) :: NFOBR, JFIRST, JLAST
        integer :: K

        do K = 1, NFOBR
            if (JFIRST .EQ. JLAST) then
                PROFILE(K) = EMINT(K)
            else
                PROFILE(K) = PROFILE(K) + W*EMINT(K)
            endif
        enddo

    end subroutine COMPRO

end program test_compro